eventsourcing/src/test/scala/com/evolutiongaming/akkaeffect/eventsourcing/JournalKeeperTest.scala

```diff
@@ -92,6 +96,23 @@ class JournalKeeperTest extends AsyncFunSuite with Matchers {
     } yield {}
   }
 
+  private def `ignore non-persisted snapshot`[F[_]: Async]: F[Unit] = {
+
+    type S = F[Unit]
+
+    val config = JournalKeeper.Config(saveSnapshotPerEvents = 2, saveSnapshotCooldown = 0.millis)
+
+    for {
+      deferred <- Deferred[F, Unit]
+      actions  <- Actions.of[F, S]
+      metadata  = SnapshotMetadata(seqNr = 2, timestamp = Instant.ofEpochMilli(0), persisted = false)
+      _        <- journalKeeperOf(2, deferred.complete(()).void, metadata.some, config, actions)
+      _        <- deferred.get
+      actions  <- actions.get
+      _         = actions shouldEqual List(Action.SaveSnapshot(2))
```

Comment:
I do not understand this:
* we create snapshot's metadata at `2` and `persisted = false`
* store it
* why there is check that snapshot was saved, if it was not `persisted`?

Reply:
because snapshot was saved :) 

`JournalKeeper` has configuration
```
val config = JournalKeeper.Config(
  saveSnapshotPerEvents = 2, 
  saveSnapshotCooldown = 0.millis
)
```
so it will save snapshot after `seqNr` will be `+ 2` after last saved snapshot or `SeqNr.Min` witch is `0`. In the test there's no snapshot (one with `persisted = false` ignored) so `JournalKeeper` must immediately save new one - that's what checked in the test.

Reply:
does it mean that if we would get snapshot with `seqNr = 2` and `persisted = true`, it would save snapshot too?

I am a bit lost why it is OK to save it in both case - it is not clear for me, what this test case tests.

Reply:
> does it mean that if we would get snapshot with `seqNr = 2` and `persisted = true`, it would save snapshot too?

no it would not because metadata says that last saved snapshot has `seqNr = 2` and current `seqNr = 2` also. Thus next saved snapshot will be on `seqNr = 4` only

Reply:
oh, `SnapshotMetadata` is the description of "current" state?